### PR TITLE
Task J-187: added show modal when user try delete podcast or episode

### PR DESCRIPTION
**Pull request by vitalik-prog**
image

### CODE REVIEW

File: frontend/src/components/episode/episode.tsx [hunk 1]
```diff
@@ -121,11 +128,17 @@ const Episode: React.FC = () => {
...
                       className={styles.deleteButton}
                     >
                       <span className="visually-hidden">Delete episode</span>
                     </button>
+                    <Modal isShowModal = {showModal} onCloseModal = {handleShowModal}>
+                      <h2>Are you sure?</h2>
+                      <div className = {styles.btnWrapper}>
+                        <Button onClick = {handleDeleteEpisode} label = {'Delete'}></Button>
+                      </div>
+                    </Modal>
```
Comment (what1s1ove) [inline frontend/src/components/episode/episode.tsx]: create a `ConfirmPopup` for it

File: frontend/src/components/episode/episode.tsx [hunk 2]
```diff
@@ -121,11 +128,17 @@ const Episode: React.FC = () => {
...
-                      onClick={handleDeleteEpisode}
+                      onClick={handleShowModal}
                       className={styles.deleteButton}
                     >
                       <span className="visually-hidden">Delete episode</span>
                     </button>
+                    <Modal isShowModal = {showModal} onCloseModal = {handleShowModal}>
+                      <h2>Are you sure?</h2>
+                      <div className = {styles.btnWrapper}>
+                        <Button onClick = {handleDeleteEpisode} label = {'Delete'}></Button>
```
Comment (what1s1ove) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
                        <Button onClick = {handleClose} label="Cancel"></Button>
                        <Button onClick = {handleDeleteEpisode} label="Delete"></Button>
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/common/confirm-popup/confirm-popup.tsx [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import { Button, Modal } from 'components/common/common';
+import styles from './styles.module.scss';
+
+type Props = {
+  isShowModal: boolean;
+  onCloseModal: () => void;
+  onDelete: () => void;
```
Comment (what1s1ove) [inline frontend/src/components/common/confirm-popup/confirm-popup.tsx]:
```suggestion
  isOpen: boolean;
  onClose: () => void;
  onConfigrm: () => void;
```

File: frontend/src/components/common/confirm-popup/confirm-popup.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+};
+
+const ConfirmPopup: React.FC<Props> = ({
+  isShowModal = true,
```
Comment (what1s1ove) [inline frontend/src/components/common/confirm-popup/confirm-popup.tsx]:
```suggestion
  isShowModal,
```

File: frontend/src/components/common/confirm-popup/confirm-popup.tsx [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  onCloseModal,
+  onDelete,
+}) => {
+  return (
+    <Modal isShowModal={isShowModal} onCloseModal={onCloseModal}>
+      <div className={styles.popup}>
+        <h2>Are you sure?</h2>
+        <div className={styles.btnWrapper}>
+          <Button onClick={onDelete} label="Delete"></Button>
+          <Button onClick={onCloseModal} label="Cancel"></Button>
```
Comment (what1s1ove) [inline frontend/src/components/common/confirm-popup/confirm-popup.tsx]:
```suggestion
          <Button onClick={onDelete} label="Confirm" />
          <Button onClick={onCloseModal} label="Cancel" />
```

File: frontend/src/components/common/confirm-popup/confirm-popup.tsx [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+const ConfirmPopup: React.FC<Props> = ({
+  isShowModal = true,
+  onCloseModal,
+  onDelete,
+}) => {
+  return (
+    <Modal isShowModal={isShowModal} onCloseModal={onCloseModal}>
+      <div className={styles.popup}>
+        <h2>Are you sure?</h2>
```
Comment (what1s1ove) [inline frontend/src/components/common/confirm-popup/confirm-popup.tsx]:
```suggestion
        <h2>{title}</h2>
        <p>{description}</p>
```
add new prop

File: frontend/src/components/episode/episode.tsx [hunk 7]
```diff
@@ -59,6 +60,8 @@ const Episode: React.FC = () => {
     dispatch(episodeActions.loadEpisodePayload(Number(id)));
   }, []);
 
+  const [showModal, setShowModal] = useState(false);
```
Comment (what1s1ove) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
  const [isConfirmPopupOpen, setIsConfirmPopupOpen] = useState<boolean>(false);
```

File: frontend/src/components/podcast/podcast.tsx [hunk 8]
```diff
@@ -28,6 +28,8 @@ const Podcast: React.FC = () => {
     dispatch(podcastActions.loadEpisodesByPodcastId(Number(id)));
   }, []);
 
+  const [showModal, setShowModal] = useState(false);
```
Comment (what1s1ove) [inline frontend/src/components/podcast/podcast.tsx]:
```suggestion
  const [isConfirmPopupOpen, setIsConfirmPopupOpen] = useState<boolean>(false);
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: frontend/src/components/episode/episode.tsx [hunk 9]
```diff
@@ -121,11 +128,16 @@ const Episode: React.FC = () => {
...
-                      onClick={handleDeleteEpisode}
+                      onClick={handleShowModal}
                       className={styles.deleteButton}
                     >
                       <span className="visually-hidden">Delete episode</span>
                     </button>
+                    <ConfirmPopup
+                      isShowModal={showModal}
+                      onCloseModal={handleShowModal}
+                      onDelete={handleDeleteEpisode}
```
Comment (what1s1ove) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
                    <ConfirmPopup
                      title="Delete Episode"
                      description="You are going to delete the episode. Are you sure about this?"
                      isShowModal={showModal}
                      onCloseModal={handleShowModal}
                      onDelete={handleDeleteEpisode}
```

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by what1s1ove — APPROVED** [review]

**Review by JuliaDzyuba — APPROVED** [review]

**Pull request merged**